### PR TITLE
Az --help generator and fix bug on buildpackage.ps1

### PR DESCRIPTION
**Pull request by arroyc**

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@arroyc__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (arroyc). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by arroyc** [comment]
we need to have a help file "az.hlp" for now in the drop/.../pkg directory to have that "az --help to work"

File: tools/CLU/BuildPackage.ps1 [hunk 1]
```diff
@@ -35,9 +35,9 @@ if ($packageId -ne "Microsoft.CLU.Commands")
 {
     $contentFileText += "    <file src=""content\*.lx"" target=""content""/>`r`n"
     $contentFileText += "    <file src=""content\*xml"" target=""content""/>`r`n"
-    if (Test-Path "$packageSource\content\help") 
+    if (Test-Path "$packageSource\content\content\help")
```
Comment (markcowl) [inline tools/CLU/BuildPackage.ps1]: @arroyc  Why add another layer?  We should just be searching in content/help

File: tools/CLU/BuildPackage.ps1 [hunk 2]
```diff
@@ -35,9 +35,9 @@ if ($packageId -ne "Microsoft.CLU.Commands")
 {
     $contentFileText += "    <file src=""content\*.lx"" target=""content""/>`r`n"
     $contentFileText += "    <file src=""content\*xml"" target=""content""/>`r`n"
-    if (Test-Path "$packageSource\content\help") 
+    if (Test-Path "$packageSource\content\content\help") 
     {
-        $contentFileText += "    <file src=""content\help\*.hlp"" target=""content\help""/>`r`n"
+		$contentFileText += "    <file src=""content\content\help\*hlp"" target=""content\help""/>`r`n"
     }
```
Comment (markcowl) [inline tools/CLU/BuildPackage.ps1]: @arroyc same comment

File: src/CLU/Microsoft.CLU/CommandLineParser/UnixCommandLineParser.cs [hunk 3]
```diff
@@ -195,11 +195,33 @@ private void PresentCommandCompletion(string[] arguments)
...
+            // Static help implementation of "az --help"
             if (arguments.Length == 0)
             {
-                // TODO
-                Console.WriteLine("Present help for Azure itself...");
+                System.IO.DirectoryInfo di = new System.IO.DirectoryInfo(CLUEnvironment.GetPackagesRootPath());
+                String rootPath = CLUEnvironment.GetPackagesRootPath();
+                try
+                {
+                    var files = from file in System.IO.Directory.EnumerateFiles(@rootPath, "az.hlp", System.IO.SearchOption.AllDirectories)
```
Comment (markcowl) [inline src/CLU/Microsoft.CLU/CommandLineParser/UnixCommandLineParser.cs]: nit: line breaks and indentation

File: tools/CLU/BuildPackage.ps1 (hunk 1 above)
Comment (arroyc) [inline tools/CLU/BuildPackage.ps1]: When I try to build it ... I don't see those help files getting copied  to drop/... . I found out it's creating the help files in src\bin\...\publish\..\content\content\help. It was more like a fix than a intentional change. Can you please try the previous version of this script and see if that was able to copy the help files properly?

File: tools/CLU/BuildPackage.ps1 (hunk 2 above)
Comment (arroyc) [inline tools/CLU/BuildPackage.ps1]: same ans

**Comment by markcowl** [comment]
@azuresdkci retest this please

File: src/CLU/Microsoft.Azure.Commands.Websites/Content/help/az.hlp [hunk 4]
```diff
@@ -0,0 +1,4 @@
+
```
Comment (markcowl) [inline src/CLU/Microsoft.Azure.Commands.Websites/Content/help/az.hlp]: @arroyc as we discussed yesterday Replace this please. with something like welcome to the azure management commands.  Run 'azure help' to get a list of all commands."

**Comment by markcowl** [comment]
@arroyc  merging for now, but please file an item to add tests for the completion functionality

File: src/CLU/Microsoft.Azure.Commands.Websites/Content/help/az.hlp (hunk 4 above)
Comment (arroyc) [inline src/CLU/Microsoft.Azure.Commands.Websites/Content/help/az.hlp]: Thanks @markcowl  .. I am working on the tests ...

**Pull request merged**